### ISSUE TITLE
NumberOverflow error when parsing JSON with big numbers on 32bit devices

### ISSUE THREAD

**Issue by siuying** [bignerdranch/Freddy#159]
Consider following test case:

```swift
    func testBigInteger() {
        let str = "{\"id\": 598458199574970368, \"id_str\": \"598458199574970368\"}"
        let data = str.dataUsingEncoding(NSUTF8StringEncoding)!
        let json = try? NSJSONSerialization.JSONObjectWithData(data, options: [])
        XCTAssertEqual(json!["id_str"], "598458199574970368")
        
        let json2 = try? JSON(data: data)
        XCTAssertEqual(try json2!.string("id_str"), "598458199574970368")
    }
```

If I run it on 32 bit devices (e.g. iPhone 4S simulator), it will fail. Note that I didn't try to access the big integer "id". The parser fail even we only try to access "id_str" which is string.

**Comment by jgallagher** [comment]
Thanks for the report. I believe this is a dupe of #76, and will be fixed once #152 (or something similar) is merged in.

**Comment by siuying** [comment]
Sorry for the dup! I just searched the error and didn't see it! Thanks for the update.
